ttnn/cpp/ttnn/operations/creation.hpp

```diff
@@ -173,11 +171,36 @@ struct FullLikeWith {
 
 struct ZerosLike : FullLikeWith<0.0f> {};
 struct OnesLike : FullLikeWith<1.0f> {};
-struct EmptyLike : FullLikeWith<0.0f> {};
 
 inline constexpr ZerosLike zeros_like{};
 inline constexpr OnesLike ones_like{};
-inline constexpr EmptyLike empty_like{};
+
+struct Empty {
+   static ttnn::Tensor invoke(
+    const ttnn::Shape& shape,
+    const DataType& dtype,
+    const Layout& layout,
+    Device* device,
+    const std::optional<MemoryConfig>& memory_config = ttnn::DRAM_MEMORY_CONFIG) {
+        TT_FATAL(memory_config.has_value(), "MemoryConfig cannot be null");
+        return create_device_tensor(shape, dtype, layout, device, memory_config.value());
```

Comment:
Minor: Since we asserted that mem config has a value immediately before, we should use `*memory_config` instead of `.value()` since `.value()` also performs the same check